src/db_adapters/user.py

```
@@ -0,0 +1,41 @@
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from src.db_tables.user import User
+from src.db_tables.user_list import UserList, UserListKind
+from src.utils.iterators import get_first
+
+
+async def user_get(session: AsyncSession, discord_id: int) -> User | None:
+    """Get a user by their Discord ID."""
+    return await session.get(User, discord_id)
+
+
+async def user_get_safe(session: AsyncSession, discord_id: int) -> User:
+    """Get a user by their Discord ID, creating them if they don't exist."""
+    user = await user_get(session, discord_id)
+    if user is None:
+        user = User(discord_id=discord_id)
+        session.add(user)
+        await session.commit()
+
+    return user
+
+
+async def user_create_list(session: AsyncSession, user: User, name: str, item_kind: UserListKind) -> UserList:
+    """Create a new list for a user."""
+    user_list = UserList(user_id=user.discord_id, name=name, item_kind=item_kind)
+    session.add(user_list)
+    await session.commit()
+    await session.refresh(user, ["lists"])
+
+    return user_list
```

What if a list with this name already exists?

(I think it would just raise some sqlalchemy exception, probably integrity error or something like that, should we let that propagate up, or catch it and raise something else? Even if we're just gonna let i propagate, I'd like the error to be documented in the doctring, like `:raises: ...`.

Regardless of whether we'll use a custom exception, or just document the sqlalchemy one, it should be done consistently everywhere where this can occur (not just in this function).